Applications/ApplicationsLib/TestDefinition.cpp

```diff
@@ -0,0 +1,229 @@
+/**
+ * \file
+ *
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "TestDefinition.h"
+
+#include <cstdlib>
+
+#include "BaseLib/ConfigTree.h"
+#include "BaseLib/Error.h"
+#include "BaseLib/FileTools.h"
+#ifdef USE_PETSC
+#include "MeshLib/IO/VtkIO/VtuInterface.h"  // For petsc file name conversion.
+#include <petsc.h>
+#endif
+
+namespace
+{
+/// Safe conversion of a double to a string using decimal or decimal exponent
+/// notation. See std::snprintf() for details for "%g" conversion specifier.
```

Comment:
How about ostringstream?

Reply:
Good point. Even better, I rewrote that part to avoid any conversion. Could you please check the changes in the last commit?

Reply:
The new changes look good.